deltachat-ios/MessageKit/Views/FileView.swift

```diff
@@ -0,0 +1,90 @@
+import UIKit
+import DcCore
+
+class FileView: UIView {
+
+    static let badgeSize: CGFloat = 54
+    static let defaultHeight: CGFloat = 78
+    static let defaultWidth: CGFloat = 250
+
+    private lazy var titleView: MessageLabel = {
+        let label = MessageLabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var subtitleView: MessageLabel = {
+        let label = MessageLabel()
+        label.numberOfLines = 1
+        label.adjustsFontSizeToFitWidth = false
+        label.lineBreakMode = .byTruncatingTail
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var fileBadgeView: InitialsBadge = {
+        let badge: InitialsBadge = InitialsBadge(image: UIImage(), size: FileView.badgeSize)
+        badge.isAccessibilityElement = false
+        badge.isHidden = false
+        badge.cornerRadius = 6
+        return badge
+    }()
+
+    private lazy var verticalStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [titleView, subtitleView])
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.alignment = .leading
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.directionalLayoutMargins = NSDirectionalEdgeInsets(top: FileMessageCell.insetHorizontalSmall,
+                                                                     leading: FileMessageCell.insetHorizontalSmall,
+                                                                     bottom: FileMessageCell.insetHorizontalSmall,
+                                                                     trailing: FileMessageCell.insetHorizontalSmall)
+        return stackView
+    }()
+
+    init() {
+        super.init(frame: .zero)
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubview(fileBadgeView)
+        addSubview(verticalStackView)
+        addConstraints([
+            fileBadgeView.constraintAlignLeadingTo(self),
+            fileBadgeView.constraintWidthTo(FileView.badgeSize),
+            fileBadgeView.constraintHeightTo(FileView.badgeSize),
+            fileBadgeView.constraintCenterYTo(self)
+        ])
+        addConstraints([
+            verticalStackView.constraintCenterYTo(self),
+            verticalStackView.constraintToTrailingOf(fileBadgeView),
+            verticalStackView.constraintAlignTrailingTo(self)
+        ])
```

Comment:
We have two subviews, both do only centerY-align, but neither sets top- or bottom-anchors. 
The point is, if we want the corresponding to be able to self-size, we need to specify these anchors. I think since we are using labels whose size is dependent of it's content, we should try to make this self sizable.

When using a stackview it is a bit tricky because we need to lower priorities for that.

Currently short document titles are well centered, but the current layout struggles a bit with longer document titles.

Reply:
> Currently short document titles are well centered, but the current layout struggles a bit with longer document titles.

This is unrelated to the stack view though. All cells in MessageKit are not self sizing in the 'normal' way. Instead there are the message size calculators that define the size (which is one of the weaknesses of the framework in my opinion). So adding Bottom or Top anchors won't change anything currently.

Reply:
ah true. forgot about that it's done by the size calculator.

Reply:
fixing this issue would also help: https://github.com/deltachat/deltachat-ios/issues/741